pkg/metadata/scheduler.go

```diff
@@ -86,6 +86,11 @@ func (c *Scheduler) AddCollector(name string, interval time.Duration) error {
 	}
 
 	go func() {
+		firstRun := &time.Timer{}
+		if withFirstRun, ok := p.(CollectorWithFirstRun); ok {
+			firstRun = newTimer(withFirstRun.FirstRunInterval())
+		}
```

Comment:
Out of curiosity, why not just use a different initial duration for the existing timer ? It would avoid having to handle two timers.
The only drawback I see is that we can't log a different error if the first one fails...

```go
initialDuration := interval
if withFirstRun, ok := p.(CollectorWithFirstRun); ok {
	initialDuration = withFirstRun.FirstRunInterval()
}
sc := &scheduledCollector{
	sendTimer:    newTimer(initialDuration),
	healthHandle: health.RegisterLiveness("metadata-" + name),
}
```

Reply:
you're right, it's much better :+1: